Sources/_StringProcessing/Regex/Core.swift

```diff
@@ -135,6 +135,25 @@ extension Regex {
   }
 }
 
+@available(SwiftStdlib 5.7, *)
+@_spi(RegexBenchmark)
+extension Regex {
+  /// Compiles the stored DSLTree into bytecode and return if it was successful
+  /// For measuring compilation times
+  ///
+  /// Note: This bypasses the cached program that is normally used
+  public func _compileRegex() -> Bool {
+    do {
+      let _ = try Compiler(
+        tree: program.tree,
+        compileOptions: program.compileOptions).emit()
+      return true
+    } catch {
+      return false
+    }
+  }
+}
```

Comment:
This kind of approach looks ok to me. I wonder if we can generalize the interface a little bit to give us some future flexibility without needing to add more SPI over time. For example, we could return a (resilient) result type which we can add metrics to in the future (number of instructions, complexity or star-height, etc). We could also have a (resilient) compilation options parameter, such as the proposed force-recompilation behavior which we could extend with more options like control over optimizations or even targeted engine version (e.g. back deployment)